add licensing notification

christinaausley opened the pull request:
## Description

Relates to https://github.com/camunda/product-hub/issues/2286.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [x] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

christinaausley commented:
@conceptualshark Want to have a glance over this and contribute anything you feel is missing?

conceptualshark commented on docs/self-managed/setup/install.md:
````diff
@@ -52,6 +52,23 @@ Refer to the [Operate](/self-managed/operate-deployment/operate-configuration.md
+  existingSecretKey:
+```
+
+This means Camunda 8 components may request valid licensing within the logs, and you may notice a "no license" label in the navigation bar. However, this has no impact on startup.
````
It looks like this is reading as `Non-Production License` in the nav bar at the moment.

conceptualshark commented on docs/self-managed/setup/install.md:
```diff
@@ -52,6 +52,23 @@ Refer to the [Operate](/self-managed/operate-deployment/operate-configuration.md
 
 When installing the [camunda-platform](https://artifacthub.io/packages/helm/camunda/camunda-platform) Helm chart, all components shown on the architectural diagram above are installed.
 
+## Licensing
```
Unsure about the positioning of this section - right now, it sits above the Helm chart instructions, and isn't necessary/a prerequisite for installation. What if this sat beneath the Installation section and above [Configuring Enterprise components and Connectors](https://docs.camunda.io/docs/next/self-managed/setup/install/#configuring-enterprise-components-and-connectors)? The page outline would look like prereqs > installation > configure license > configure components.

christinaausley replied:
Adjusted in latest commit -- I am indeed wondering if we shouldn't add something to announcements if we are worried users might miss this, or wonder why the nav/logs are showing this now. Do you have any strong opinions?

christinaausley replied:
I added throughout the components in Self Managed, but let me know if you feel this is overkill.

conceptualshark commented on docs/self-managed/setup/install.md:
````diff
@@ -52,6 +52,23 @@ Refer to the [Operate](/self-managed/operate-deployment/operate-configuration.md
+  existingSecretKey:
+```
+
+This means Camunda 8 components may request valid licensing within the logs, and you may notice a "no license" label in the navigation bar. However, this has no impact on startup.
````
```suggestion
Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup.
```

I know this change is in /next, but do you think it's worth calling out somehow that licenses don't/won't exist for enterprise customers yet? My concern is someone reading these docs or seeing the warning and worrying they are an enterprise customer and have no license, but I'm not sure how valid that fear is.

christinaausley replied:
Agreed on these changes 👍

conceptualshark reviewed (commented):
I think a form of these changes should likely also be replicated in the [Docker guide](https://docs.camunda.io/docs/next/self-managed/setup/deploy/other/docker/#configuration-hints) - we can document the `CAMUNDA_LICENSE_KEY` env var here ([slack discussion](https://camunda.slack.com/archives/C06V2SR6U9G/p1722257808134929?thread_ts=1722257549.082199&cid=C06V2SR6U9G)) so it's at least searchable in some fashion.

christinaausley reviewed (commented):
Note to self that we would still need to add a note to the upgrade documentation.

conceptualshark commented:
Not settled yet on if the redundancy of the "Non-Production License" warning is too much in the components.  🤔 I pushed some changes to this based on my understanding - trying to explain the use without implying people need to start doing it Now.

theburi commented on docs/self-managed/console-deployment/configuration.md:
```diff
@@ -17,21 +17,26 @@ Underscores in environment variables correspond to configuration file key levels
 Console environment variables could be set in Helm via the `console.env` key. For more details, check [Console Helm values](https://artifacthub.io/packages/helm/camunda/camunda-platform#console-parameters).
 
+:::note
+Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup.
```
I think we can call that key will not affect functionality of Console
'have no impact on startup or functionality.'

christinaausley replied:
Adjusted in latest commit 👍

theburi commented on docs/self-managed/modeler/web-modeler/configuration/configuration.md:
```diff
@@ -10,6 +10,20 @@ The different components of Web Modeler Self-Managed can be configured using env
+For Helm installations, license keys can be configured globally in your `values.yaml` file. See the [Helm installation documentation](/self-managed/setup/install.md#configure-license-key) for more details.
+
+:::note
+Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup.
```
We should mention 5 user limit or reference limitations in some other source

christinaausley replied:
Five user limit if you do not have a valid license? When you say "in some other source", do you mean licensing documentation?

christinaausley replied:
@theburi Following up here if you'd like me to add this 👍

aabouzaid reviewed (approved):
LGTM, thank you :+1:

christinaausley commented:
This is ready for final review, keeping in mind we will need to update the update documentation, pun unintended.

MaxTru commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,34 @@ Scheduled release date: 8th of Oct 2024
 
 Scheduled end of maintenance: 14th of April 2026
 
+### Configuring license keys
```
I propose to keep the announcement short and focus on:
1. With 8.6 Camunda requires a license key to run in production (link to blog post)
2. Please find out here how to set the license key (link to install instructions for sm)

christinaausley replied:
Adjusted in latest commit 👍

MaxTru reviewed (changes requested):
One small suggestion regarding the general setup.

theburi commented on docs/self-managed/console-deployment/configuration.md:
```diff
@@ -17,21 +17,26 @@ Underscores in environment variables correspond to configuration file key levels
 Console environment variables could be set in Helm via the `console.env` key. For more details, check [Console Helm values](https://artifacthub.io/packages/helm/camunda/camunda-platform#console-parameters).
 
+:::note
+Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup or functionality.
```
Can we add a note here and other components? Something in line with: `To obtain a license, please visit https://camunda.com/platform/camunda-platform-enterprise-contact/`

christinaausley replied:
Added 👍

RomanJRW commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,34 @@ Scheduled release date: 8th of Oct 2024
+- Enter your license key directly in `global.license.key`
+- Provide a secret name and secret key in `global.license.existingSecret` and `global.license.existingSecretKey`
+
+If your installation of Camunda 8 requires a license key, it can be provided to **each component** as an environment variable. See the documentation for your components for more information.
```
Minor, but the repeated `If your installation of Camunda 8 requires a license key` reads a bit weird to me. What about instead adding a third bullet point above with simply `See the documentation for your components for more information on how to provide the license key to each component`

christinaausley replied:
Adjusted 👍

RomanJRW commented on optimize/self-managed/optimize-deployment/configuration/system-configuration-platform-8.md:
```diff
@@ -13,6 +13,22 @@ description: "Connection to Camunda 8."
+
+:::note
+Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup or functionality.
+:::
```
On this page: https://docs.camunda.io/docs/next/self-managed/setup/deploy/other/docker/#optimize

There is a property called `CAMUNDA_OPTIMIZE_ENTERPRISE`. I believe that this would now be redundant in C8. Shall we remove it there? I cannot find equivalent pages for the other components

conceptualshark replied:
All set here: https://github.com/camunda/camunda-docs/issues/4131

christinaausley commented:
Hi all -- I have incorporated your feedback and converted some of the repetitive text into a React component so we only need to make changes across a few places. Let me know if you have anything else to add 👍

conceptualshark reviewed (dismissed):
Thanks for all the work on this! 🙌

MaxTru commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,16 @@ Scheduled release date: 8th of Oct 2024
 
 Scheduled end of maintenance: 14th of April 2026
 
+### Configuring license keys
```
I would call this something like "License key change"

MaxTru commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,16 @@ Scheduled release date: 8th of Oct 2024
 
+### Configuring license keys
+
+Camunda 8 components are now able to consume Enterprise licensing information. For additional details, review the [blog post on licensing updates for Camunda 8 Self-Managed](https://camunda.com/blog/2024/04/licensing-update-camunda-8-self-managed/).
```
I am thinking:

```suggestion
With the 8.6 release, Camunda 8 Self-Managed requires a license key for production usage. For additional details, review the [blog post on licensing updates for Camunda 8 Self-Managed](https://camunda.com/blog/2024/04/licensing-update-camunda-8-self-managed/).
```

christinaausley replied:
Adjusted in latest commit 👍

MaxTru commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,16 @@ Scheduled release date: 8th of Oct 2024
+
+Camunda 8 components are now able to consume Enterprise licensing information. For additional details, review the [blog post on licensing updates for Camunda 8 Self-Managed](https://camunda.com/blog/2024/04/licensing-update-camunda-8-self-managed/).
+
+Review the documentation for your components for more information on how to provide the license key to each component as an environment variable. To configure with Helm, visit the [Self Managed installation documentation](/self-managed/setup/install.md).
```
Would it make sense to link all the sub-pages from here for easier discovery?

christinaausley replied:
Added in latest commit 👍

theburi commented on docs/reference/announcements.md:
```diff
@@ -10,6 +10,16 @@ Scheduled release date: 8th of Oct 2024
+Review the documentation for your components for more information on how to provide the license key to each component as an environment variable. To configure with Helm, visit the [Self Managed installation documentation](/self-managed/setup/install.md).
+
+:::note
+Camunda 8 components without a valid license may display **Non-Production License** in the navigation bar and issue warnings in the logs. These warnings have no impact on startup or functionality. To obtain a license, visit the [Camunda Enterprise page](https://camunda.com/platform/camunda-platform-enterprise-contact/).
```
Could we amend this line, please? The overall announcement is not 100% correct since Web Modeller has a limitation of 5 users: "These warnings have no impact on startup or functionality."
Maybe be more accurate to link to a https://camunda.com/pricing/ page? We don't have a dedicated license page.

aabouzaid reviewed (dismissed):
LGTM, thank you :+1:

I pushed a small enhancement to have a fully working example for adding the license.

christinaausley commented:
Merging this in given the review in this PR -- any additional changes should be made in a separate PR 👍
